Sources/XMTPiOS/Client.swift

```diff
@@ -100,23 +101,40 @@ public final class Client: Sendable {
 	}
 
 	static func create(account: SigningKey, apiClient: ApiClient, options: ClientOptions? = nil) async throws -> Client {
-		let privateKeyBundleV1 = try await loadOrCreateKeys(for: account, apiClient: apiClient, options: options)
+		let (privateKeyBundleV1, source) = try await loadOrCreateKeys(for: account, apiClient: apiClient, options: options)
+
+		let dbURL = URL.documentsDirectory.appendingPathComponent("xmtp-\(account.address).db3")
+		let v3Client = try await LibXMTP.createClient(
+			logger: XMTPLogger(),
+			host: GRPCApiClient.envToUrl(env: apiClient.environment),
+			isSecure: apiClient.environment != .local,
+			db: dbURL.path,
+			encryptionKey: nil,
+			accountAddress: account.address,
+			legacyIdentitySource: source,
+			legacySignedPrivateKeyProto: nil
```

Comment:
I don't think this should be nil?

Reply:
yea I misunderstood what it's from. Updated to pass that in.